### PR TITLE
Combining dockerfiles

### PR DESCRIPTION
**Pull request by shimwell**
## Proposed changes

This PR removes the need for two seperate Dockerfiles (Dockerfile and DockerfileDependencies) by using a two stage build as suggested by @agoose77 in issue https://github.com/ukaea/paramak/issues/745

This is nice as it reduces maintenance of two dockerfiles and simplifies the building and testing process.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/dockerhub-publish-dependencies.yml [hunk 1]
```diff
@@ -31,10 +31,10 @@ jobs:
         id: docker_build
         uses: docker/build-push-action@v2
         with:
-          file: DockerfileDependencies
           push: true
           tags: ukaea/paramak:dependencies
           build-args: |
+            target=dependencies
```
Comment (agoose77) [inline .github/workflows/dockerhub-publish-dependencies.yml]: I think this needs to be put in the parent section, as a sibling to `tags`:
https://github.com/docker/build-push-action#inputs

**Comment by agoose77** [comment]
Thank you for adding me as a contributor!

File: .github/workflows/dockerhub-publish-dependencies.yml (hunk 1 above)
Comment (shimwell) [reply .github/workflows/dockerhub-publish-dependencies.yml]: ah yes I see, I had mixed it up with the build args incorrectly. I've moved it to tags and we can always merge and see if the action works. It triggers with every push to develop so we will find out one way or the other

**Comment by agoose77** [comment]
I think it might need to be up one more level:
```yaml
        with:
          push: true
          tags: 
            - ukaea/paramak:dependencies
          target: dependencies
          build-args:
            - cq_version=master
            - compile_cores=2
            - include_neutronics=true
```

**Comment by shimwell** [comment]
I think this might have been done correctly this time, sorry for misunderstanding. Happy to merge and see if the actions work

**Review by RemDelaporteMathurin — APPROVED** [review]
This is a great improvement! thanks @agoose77 for the suggestion!!

**Pull request merged**